director/profile.go

```diff
@@ -323,27 +324,30 @@ func SavedDeviceProfileDiffers(device types.Device, profile types.DeviceProfile)
 			InfoLogger(LogHolder{DeviceUDID: device.UDID, DeviceSerial: device.SerialNumber, ProfileIdentifier: profile.PayloadIdentifier, ProfileUUID: profile.HashedPayloadUUID, Message: "Hashed payload UUID doesn't match what's in ProfileList", Metric: profileList.PayloadUUID})
 		}
 		// May be waiting for a device to report in full - just bail if there profilelist count is 0
-		// var profileCount int
-		// err := db.DB.Model(&profileList).Where("device_ud_id = ? AND payload_identifier = ?", device.UDID, profile.PayloadIdentifier).Count(&profileCount).Error
-		// if err != nil {
-		// 	if !gorm.IsRecordNotFoundError(err) {
-		// 		// If it's not found, we'll catch in the false return at the end. Else raise an error
-		// 		return true, errors.Wrap(err, "Could not load ProfileList for device")
-		// 	}
-		// }
-		// if profileCount == 0 {
-		// 	InfoLogger(LogHolder{DeviceUDID: device.UDID, DeviceSerial: device.SerialNumber, ProfileIdentifier: profile.PayloadIdentifier, ProfileUUID: profile.HashedPayloadUUID, Message: "Device has an empty ProfileList stored"})
-		// }
-		// skipCommands := []string{"ProfileList", "SecurityInfo", "DeviceInformation", "CertificateList"}
-		// tenMinsAgo := time.Now().Add(-10 * time.Minute)
-		// for _, item := range skipCommands {
-		// 	inQueue := CommandInQueue(device, item, tenMinsAgo)
-		// 	if !inQueue {
-		// 		InfoLogger(LogHolder{DeviceUDID: device.UDID, DeviceSerial: device.SerialNumber, ProfileIdentifier: profile.PayloadIdentifier, ProfileUUID: profile.HashedPayloadUUID, Message: "Requesting Device Info", Metric: profileList.PayloadUUID})
-		// 		_ = RequestAllDeviceInfo(device)
-		// 		break
-		// 	}
-		// }
+		var profileCount int
+		err := db.DB.Model(&profileList).Where("device_ud_id = ?", device.UDID).Count(&profileCount).Error
+		if err != nil {
+			if !gorm.IsRecordNotFoundError(err) {
+				// If it's not found, we'll catch in the false return at the end. Else raise an error
+				return true, errors.Wrap(err, "Could not load ProfileList for device")
+			}
+		}
+		if profileCount == 0 {
+			InfoLogger(LogHolder{DeviceUDID: device.UDID, DeviceSerial: device.SerialNumber, ProfileIdentifier: profile.PayloadIdentifier, ProfileUUID: profile.HashedPayloadUUID, Message: "Device has an empty ProfileList stored"})
+		}
+		skipCommands := []string{"ProfileList", "SecurityInfo", "DeviceInformation", "CertificateList"}
```

Comment:
Why is this variable named `skipCommands`? Because if any of these commands are in the queue we will skip `RequestAllDeviceInfo()` now?

Reply:
Correct